### ISSUE TITLE
Having a bit of an issue with the DSL

### ISSUE THREAD

**Issue by bararchy** [at-grandpa/clim#28]
So, I configured this: 

```crystal
class CLI < Clim
    main_command do
      desc "NexPloit CLI utility for automation and testing"
      usage "./nexploit_cli [sub_command] [arguments]"
      run do |options, arguments|
        puts options.help
      end
      sub_command "info" do
        desc "Get info from the engine instance"
        usage "./nexploit_cli info [arguments]"
        option "--uuids", type: Bool, desc: "Get all UUIDs of running scans", default: false
        option "--url", type: String, desc: "URL of the engine API", required: true

        run do |options, arguments|
          raise "No URL given in options! #{options}" unless options.url
          api = API.new(options.url.to_s)
          api.get_uuids(options.url.to_s) if options.uuids
        end
      end
      sub_command "scan" do
        desc "Initiate Scan of a new target"
        usage "./nexploit_cli scan [arguments]"
        run do |options, arguments|
          puts "Fake Crystal tool!!"
        end
      end
    end
  end
```

Now, I would expect 2 things.
First, `options.url` to always be String not String? because it's `required` else it should raise.
Second, not sure why but the `options.url` is empty (Nil) even though the argument clearly shows the var populated `["http://1.1.1.1"]`

**Comment by at-grandpa** [comment]
@bararchy 
Thank you for opening the issue.

> First, options.url to always be String not String? because it's required else it should raise.

You're right. I'll fix this problem. Thanks.


> Second, not sure why but the options.url is empty (Nil) even though the argument clearly shows the var populated ["http://1.1.1.1"]

This is due to the specification of [OptionParser#on](https://crystal-lang.org/api/0.24.2/OptionParser.html#on%28flag%3AString%2Cdescription%3AString%2C%26block%3AString-%3E%29-instance-method).
You can retrieve the value by specifying an optional argument like `--url URL`.

For example:

_option_parser_test.cr_

```crystal
require "option_parser"

OptionParser.parse! do |parser|
  parser.on("--url1", "No optional argument.") { |url| p url }
  parser.on("--url2 URL2", "Optional argument exists.") { |url| p url }
end
```

```console
$ crystal run src/option_parser_test.cr -- --url1 http://foo.com
""
$ crystal run src/option_parser_test.cr -- --url2 http://foo.com
"http://foo.com"
```

**Comment by at-grandpa** [comment]
@bararchy 

I fixed it [`v0.3.0`](https://github.com/at-grandpa/clim/tree/v0.3.0) .

**Comment by bararchy** [comment]
@at-grandpa thanks! I'll update and test
